content/spire/try/spire+aws-oidc-federation.md

```diff
@@ -0,0 +1,334 @@
+
+1. Navigate to the AWS [Identity and Access Management (IAM) page](https://console.aws.amazon.com/iam/home#/home), logging in if necessary.
+
+2. Open the IAM role that you created in the previous section, `oidc-federation-test-role`. If needed, click **Roles** on the left, use the search field to find the IAM role that you created in the last section, and click the role.
+
+3. Click the **Trust relationships** tab near the middle of the page and then click **Edit trust relationship**.
+
+4. In the JSON access control policy, add a condition line at the end of the `StringEquals` attribute to restrict access to workloads matching the workload SPIFFE ID that was assigned in the [Kubernetes Quickstart](/spire/try/getting-started-k8s/).
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::012345678901:oidc-provider/MY_DISCOVERY_DOMAIN"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "MY_DISCOVERY_DOMAIN:aud": "mys3",
+          "MY_DISCOVERY_DOMAIN:sub": "spiffe://example.org/ns/default/sa/default"
+        }
+      }
+    }
+  ]
+}
+```
+
+5. Click **Update Trust Policy**.
+
+## Create an AWS S3 Test File
+
+Create a simple test file in an AWS S3 bucket to use for testing.
+
+1. Create a text file on your local computer called 'test.txt` with the following single line:
+```
+oidc-tutorial file
+```
+You'll upload this file to the S3 bucket later.
+
+2. Navigate to the AWS [Amazon S3 page](https://s3.console.aws.amazon.com/s3/home) logging in if necessary.
+
+3. Click **Create bucket**.
+
+4. Under **Bucket name** type `oidc-federation-test-bucket`.
+
+5. Leave the **Region**, **Bucket settings for Block Public Access**, and **Advanced settings** at the default values and click **Create bucket**.
+
+6. Click `oidc-federation-test-bucket` to open the bucket. 
+
+7. Click **Upload**.
+
+8. Add the `test.txt` file to the upload area using your local file navigator or drag and drop.
+
+9. Click **Upload**.
+
+# Part 3: Test Access to AWS S3
+
+Now that Kubernetes and AWS are configured for OIDC federation, we'll test the connection. This test retrieves a JWT SVID from the SPIRE Agent and uses the JWT token in the JWT SVID to access S3.
+
+1. Run: 
+```console
+$ kubectl get pods
+
+NAME                      READY   STATUS    RESTARTS   AGE
+client-74d4467b44-7nrs2   1/1     Running   0          27s
+```
+
+2. Start a shell on the client using the client pod name just shown.
+
+```console
+$ kubectl exec -it client-74d4467b44-7nrs2 /bin/sh
+```
+
+3. Fetch a JWT SVID from the identity provider on AWS and save the token from the JWT SVID into a file called `token`:
+
+```console
+/opt/spire/bin/spire-agent api fetch jwt -audience mys3 -socketPath /run/spire/sockets/agent.sock | sed '2q' | sed 's/[[:space:]]//g' > token
```

Comment:
can you update this command to:
```
/opt/spire/bin/spire-agent api fetch jwt -audience mys3 -socketPath /run/spire/sockets/agent.sock | sed '2!d' | sed 's/[[:space:]]//g' > token
```
I found previous command was adding token ID and we didnt want that:

```
token(spiffe://example.org/ns/default/sa/default):
eyJhbGciOiJFUzI1NiIsImtpZCI6Ikt3cElUd0tqSj....
```

but we want:
```
eyJhbGciOiJFUzI1NiIsImtpZCI6Ikt3cElUd0tqSj....
```

Reply:
Yes, I've updated it.